public/spec/features/resources_spec.rb

```diff
@@ -118,4 +118,84 @@
     pdf_files = Dir.glob(File.join(Dir.tmpdir, '*.pdf'))
     expect(pdf_files).to include "#{Dir.tmpdir}/text_us_paav_b_jd99_blake_carver_papers_es_carver_xml.pdf"
   end
+
+  it 'downloads PDF from archival object view' do
+    # Clean existing PDFs
+    Dir.glob(File.join(Dir.tmpdir, '*.pdf')).each { |file| File.delete(file) }
+
+    resource = create(:resource,
+      title: "Resource with Archival Object",
+      publish: true
+    )
+
+    archival_object = create(:archival_object,
+      title: "Archival Object PDF Test",
+      resource: { 'ref' => resource.uri },
+      publish: true
+    )
+
+    run_indexers
+
+    visit('/')
+    fill_in 'q0', with: archival_object.title
+    click_on 'Search'
+    click_on archival_object.title
+
+    click_on 'Download PDF'
+
+    pdf_files = Dir.glob(File.join(Dir.tmpdir, '*.pdf'))
+    expect(pdf_files.length).to eq(1)
+  end
+
+  it 'shows PDF download button only for resources and their components' do
```

Comment:
Thanks for the tests @Vchouliaras! Can you please update this test title to reflect pdf buttons are shown only for _published_ components? eg "only for resources and their published components".

Reply:
I updated the title